satpy/readers/avhrr_l1b_gaclac.py

```diff
@@ -123,7 +123,11 @@ def read_raw_data(self):
                 interpolate_coords=self.interpolate_coords,
                 creation_site=self.creation_site,
                 **self.reader_kwargs)
-            self.reader.read(self.filename)
+            if hasattr(self.filename, 'open'):
+                with self.filename.open() as fileobj:
+                    self.reader.read(str(self.filename), fileobj)
+            else:
+                self.reader.read(self.filename)
```

Comment:
Would it be possible to do this in pygac instead of satpy?

Reply:
In other words, is it possible to pass only a file object to pygac or does it need the filename as well?

Reply:
> Would it be possible to do this in pygac instead of satpy?

Absolutely, I could/would add the attribute check for `open` to `pygac.utils._file_opener`.

> In other words, is it possible to pass only a file object to pygac or does it need the filename as well?

Theoretically, the file object would be sufficient for pygac, because it can extract the name from the product header, but in some POD cases, the filename in the header is not set, so we use the provided filename as fall back. However, a `PathLike` object with an `open` method is sufficient to provide both with a single argument, see note in the PR description as example.